src/org/rascalmpl/library/Prelude.java

```diff
@@ -3348,41 +3371,59 @@ private String toBase64(InputStream src, int estimatedSize) throws IOException {
 	  return result.toString(StandardCharsets.ISO_8859_1.name());
 	}
 
-	public IString toBase64(IString in) {
+	public IString toBase64(IString in, IString charsetName) {
 	  try {
-	      InputStream bytes = new ByteBufferBackedInputStream(StandardCharsets.UTF_8.encode(in.getValue()));
-	      return values.string(toBase64(bytes, in.length() * 2));
+		Charset charset = Charset.forName(charsetName.getValue());
+	    InputStream bytes = new ByteBufferBackedInputStream(charset.encode(in.getValue()));
+	    return values.string(toBase64(bytes, in.length() * 2));
 	  } catch (IOException e) {
 	      throw RuntimeExceptionFactory.io(values.string(e.getMessage()));
 	  }
 	}
 
 	public IString toBase64(ISourceLocation file) {
-	    try (InputStream in = REGISTRY.getInputStream(file)) {
-	        return values.string(toBase64(in, 1024));
-	    }
-	    catch (IOException e) {
-	        throw RuntimeExceptionFactory.io(values.string(e.getMessage()));
-	    }
+		try (InputStream in = REGISTRY.getInputStream(file)) {
+			return values.string(toBase64(in, 1024));
+		}
+		catch (IOException e) {
+			throw RuntimeExceptionFactory.io(values.string(e.getMessage()));
+		}
 	}
-	
-	
 
 	private void fromBase64(String src, OutputStream target) throws IOException {
 	  InputStream bytes = new ByteBufferBackedInputStream(StandardCharsets.ISO_8859_1.encode(src));
 	  copy(Base64.getDecoder().wrap(bytes), target);
 	}
 
-	public IString fromBase64(IString in) {
+	public IString fromBase64(IString in, IString charset) {
 	    try {
 	        ByteArrayOutputStream result = new ByteArrayOutputStream(in.length());
 	        fromBase64(in.getValue(), result);
-	        return values.string(result.toString(StandardCharsets.UTF_8.name()));
+	        return values.string(result.toString(charset.getValue()));
 	    } catch (IOException e) {
 	        throw RuntimeExceptionFactory.io(values.string(e.getMessage()));
 	    }
 	}
 
+	public IString toBase32(IString in, IString charset, IBool includePadding) {
+		Base32 encoder = new Base32();
+		ByteBuffer buffer = Charset.forName(charset.getValue()).encode(in.getValue());
+		byte[] data = new byte[buffer.limit()];
+		buffer.get(data);
+		String encoded = encoder.encodeToString(data);
+		if (!includePadding.getValue()) {
+			encoded = encoded.replace("=", "");
+		}
+		return values.string(encoded);
+	}
+
+	public IString fromBase32(IString in, IString charset) {
+		Base32 decoder = new Base32(72, new byte[] { '\r', '\n' }, false, (byte) '=', CodecPolicy.LENIENT);
```

Comment:
shouldn't we make the block width a parameter? I might not be expecting a newline in string, even if it's long.

Reply:
We can certainly do that. I will have to check to see if the Base64 encoder has a similar feature and if so, also add a parameter there.